CrawlJob.php

```diff
@@ -54,9 +61,34 @@ public static function create(AMQPMessage $message)
         $urlToCrawl = $data['url'];
         $baseUrl = $data['base_url'];
         $blacklist = $data['blacklist'];
+        $whitelist = $data['whitelist'];
         $metadata = $data['metadata'];
 
-        return new static($urlToCrawl, $baseUrl, $blacklist, $metadata);
+        return new static($urlToCrawl, $baseUrl, $blacklist, $whitelist, $metadata);
+    }
+
+    /**
+     * Check if given url is whitelisted
+     *
+     * @param string $url
+     * @param array $whitelist
+     *
+     * @return boolean
+     */
+   public static function isUrlWhitelisted($url, array $whitelist)
```

Comment:
We could also make this method non-static so that you don't need to supply the url which is already known.